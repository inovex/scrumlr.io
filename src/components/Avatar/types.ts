export const AVATAR_SKIN_COLORS = ["Tanned", "Yellow", "Pale", "Light", "Brown", "DarkBrown", "Black"] as const;
export type AvatarSkinColor = (typeof AVATAR_SKIN_COLORS)[number];

export const AVATAR_TOP_TYPES = [
  "Eyepatch",
  "Hat",
  "Hijab",
  "LongHairBigHair",
  "LongHairBob",
  "LongHairBun",
  "LongHairCurly",
  "LongHairCurvy",
  "LongHairDreads",
  "LongHairFrida",
  "LongHairFro",
  "LongHairFroBand",
  "LongHairMiaWallace",
  "LongHairNotTooLong",
  "LongHairShavedSides",
  "LongHairStraight",
  "LongHairStraight2",
  "LongHairStraightStrand",
  "NoHair",
  "ShortHairDreads01",
  "ShortHairDreads02",
  "ShortHairFrizzle",
  "ShortHairShaggyMullet",
  "ShortHairShortCurly",
  "ShortHairShortFlat",
  "ShortHairShortRound",
  "ShortHairShortWaved",
  "ShortHairSides",
  "ShortHairTheCaesar",
  "ShortHairTheCaesarSidePart",
  "Turban",
  "WinterHat1",
  "WinterHat2",
  "WinterHat3",
  "WinterHat4",
] as const;
export type AvatarTopType = (typeof AVATAR_TOP_TYPES)[number];

export const AVATAR_CLOTHE_COLORS = [
  "Black",
  "Blue01",
  "Blue02",
  "Blue03",
  "Gray01",
  "Gray02",
  "Heather",
  "PastelBlue",
  "PastelGreen",
  "PastelOrange",
  "PastelRed",
  "PastelYellow",
  "Pink",
  "Red",
  "White",
] as const;
export type AvatarClotheColor = (typeof AVATAR_CLOTHE_COLORS)[number];

export const AVATAR_GRAPHIC_TYPES = ["Bat", "Cumbia", "Deer", "Diamond", "Hola", "Pizza", "Resist", "Selena", "Bear", "SkullOutline", "Skull"] as const;
export type AvatarGraphicType = (typeof AVATAR_GRAPHIC_TYPES)[number];

export const AVATAR_CLOTHE_TYPES = ["BlazerShirt", "BlazerSweater", "CollarSweater", "GraphicShirt", "Hoodie", "Overall", "ShirtCrewNeck", "ShirtScoopNeck", "ShirtVNeck"] as const;
export type AvatarClotheType = (typeof AVATAR_CLOTHE_TYPES)[number];

export const AVATAR_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "PastelPink", "Blue", "Platinum", "Red", "SilverGray"] as const;
export type AvatarHairColor = (typeof AVATAR_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_COLORS = ["Auburn", "Black", "Blonde", "BlondeGolden", "Brown", "BrownDark", "Platinum", "Red"] as const;
export type AvatarFacialHairColor = (typeof AVATAR_FACIAL_HAIR_COLORS)[number];

export const AVATAR_FACIAL_HAIR_TYPES = ["Blank", "BeardMedium", "BeardLight", "BeardMajestic", "MoustacheFancy", "MoustacheMagnum"] as const;
export type AvatarFacialHairType = (typeof AVATAR_FACIAL_HAIR_TYPES)[number];

export const AVATAR_ACCESSORIES_TYPES = ["Blank", "Kurt", "Prescription01", "Prescription02", "Round", "Sunglasses", "Wayfarers"] as const;
export type AvatarAccessoriesType = (typeof AVATAR_ACCESSORIES_TYPES)[number];

export const AVATAR_EYE_TYPES = ["Close", "Cry", "Default", "Dizzy", "EyeRoll", "Happy", "Hearts", "Side", "Squint", "Surprised", "Wink", "WinkWacky"] as const;
export type AvatarEyeType = (typeof AVATAR_EYE_TYPES)[number];

export const AVATAR_EYEBROW_TYPES = [
  "Angry",
  "AngryNatural",
  "Default",
  "DefaultNatural",
  "FlatNatural",
  "FrownNatural",
  "RaisedExcited",
  "RaisedExcitedNatural",
  "SadConcerned",
  "SadConcernedNatural",
  "UnibrowNatural",
  "UpDown",
  "UpDownNatural",
] as const;
export type AvatarEyebrowType = (typeof AVATAR_EYEBROW_TYPES)[number];

export const AVATAR_MOUTH_TYPES = ["Concerned", "Default", "Disbelief", "Eating", "Grimace", "Sad", "ScreamOpen", "Serious", "Smile", "Tongue", "Twinkle", "Vomit"] as const;
export type AvatarMouthType = (typeof AVATAR_MOUTH_TYPES)[number];
