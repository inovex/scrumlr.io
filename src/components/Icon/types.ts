export type IconNames =
  | 'phase1-create'
  | 'phase1-communicate'
  | 'phase1-combine'
  | 'phase2-vote'
  | 'phase3-discuss'
  | 'phase3-define'
  | 'circle-arrow-left'
  | 'circle-arrow-right'
  | 'circle-add'
  | 'circle'
  | 'more'
  | 'plus'
  | 'circle-selection'
  | 'circle-selection-grey'
  | 'logout'
  | 'download'
  | 'edit'
  | 'donate'
  | 'pencil'
  | 'focus'
  | 'close'
  | 'close20'
  | 'close-circle'
  | 'feedback'
  | 'key-up'
  | 'key-down'
  | 'key-esc'
  | 'settings'
  | 'board'
  | 'trash'
  | 'chevron-left'
  | 'chevron-right'
  | 'stack-top'
  | 'stack-mid'
  | 'stack-hover'
  | 'arrow-left'
  | 'arrow-right';
