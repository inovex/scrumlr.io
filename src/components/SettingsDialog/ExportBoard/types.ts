import {Color} from "constants/colors";

export interface BoardType {
  id: string;
  name: string;
  showAuthors: boolean;
  showVoting: string;
  accessPolicy: string;
}

export interface ColumnType {
  id: string;
  name: string;
  color: Color;
  visible: boolean;
}

export interface NoteType {
  id: string;
  author: string;
  text: string;
  position: {
    column: string;
    stack: string | null;
    rank: number;
  };
}

export interface ParticipantType {
  role: string;
  user: {
    id: string;
    name: string;
  };
}

export interface VotingType {
  votes: {
    votes: number;
    votesPerNote: [];
  };
}

export interface BoardDataType {
  board: BoardType;
  columns: ColumnType[];
  notes: NoteType[];
  participants: ParticipantType[];
  votings: VotingType[];
}
