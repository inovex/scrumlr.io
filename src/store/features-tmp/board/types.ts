import {Participant} from "../../features/participants/types";
import {Request} from "../../features/requests/types";
import {Column} from "../../features/columns/types";
import {Note} from "../../features/notes/types";
import {Reaction} from "../../features/reactions/types";
import {Vote} from "../../features/votes/types";
import {Voting} from "../../features/votings/types";

export enum AccessPolicy {
  "PUBLIC" = 0,
  "BY_PASSPHRASE" = 1,
  "BY_INVITE" = 2,
}

export interface Board {
  id: string;

  name?: string;
  accessPolicy: keyof typeof AccessPolicy;
  showAuthors: boolean;
  showNotesOfOtherUsers: boolean;
  showNoteReactions: boolean;
  allowStacking: boolean;
  isLocked: boolean;
  timerStart?: Date;
  timerEnd?: Date;

  sharedNote?: string;
  showVoting?: string;
}

export type BoardActionType = {
  board: Board;
  participants: Participant[];
  requests: Request[];
  columns: Column[];
  notes: Note[];
  reactions: Reaction[];
  votes: Vote[];
  votings: Voting[];
};

export type EditBoardRequest = Partial<Omit<Board, "id">> & {passphrase?: string};

export type BoardStatus = "unknown" | "pending" | "ready" | "rejected" | "accepted" | "passphrase_required" | "incorrect_passphrase" | "too_many_join_requests" | "banned";

export interface BoardState {
  status: BoardStatus;
  data?: Board;
}
