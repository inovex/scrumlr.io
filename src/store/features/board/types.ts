import {Participant} from "../participants";
import {Request} from "../requests";
import {Column} from "../columns";
import {Note} from "../notes";
import {Reaction} from "../reactions";
import {Vote} from "../votes";
import {Voting} from "../votings";

export enum AccessPolicy {
  "PUBLIC" = 0,
  "BY_PASSPHRASE" = 1,
  "BY_INVITE" = 2,
}

export interface Board {
  id: string;

  name?: string;
  accessPolicy: keyof typeof AccessPolicy;
  showAuthors: boolean;
  showNotesOfOtherUsers: boolean;
  showNoteReactions: boolean;
  allowStacking: boolean;
  isLocked: boolean;
  timerStart?: Date;
  timerEnd?: Date;

  sharedNote?: string;
  showVoting?: string;
}

export type BoardActionType = {
  board: Board;
  participants: Participant[];
  requests: Request[];
  columns: Column[];
  notes: Note[];
  reactions: Reaction[];
  votes: Vote[];
  votings: Voting[];
};

export type BoardWithServerTimeOffset = {board: Board; serverTimeOffset: number};

export type EditBoardRequest = Partial<Omit<Board, "id">> & {passphrase?: string};

export type BoardStatus = "unknown" | "pending" | "ready" | "rejected" | "accepted" | "passphrase_required" | "incorrect_passphrase" | "too_many_join_requests" | "banned";

export interface BoardState {
  status: BoardStatus;
  data?: Board;
}
