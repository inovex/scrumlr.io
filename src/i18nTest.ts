import {join} from "path";
import i18n from "i18next";
import {initReactI18next} from "react-i18next";
import Backend from "i18next-fs-backend";

i18n
  .use(Backend)
  .use(initReactI18next)
  .init({
    backend: {
      loadPath: join(__dirname, "../public/locales/en/translation.json"),
    },
    initImmediate: false,
    lng: "en",
    debug: false,
    react: {
      useSuspense: false,
    },
  });

export default i18n;
