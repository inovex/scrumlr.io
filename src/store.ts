import { createStore, compose } from 'redux';
import { reactReduxFirebase } from 'react-redux-firebase';

import rootReducer from './reducers';
import { firebase as fbConfig } from './config';
import * as firebase from 'firebase';

export default function configureStore(initialState: {}, history?: {}) {
  firebase.initializeApp(fbConfig);

  const composeEnhancers =
    (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const createStoreWithMiddleware = composeEnhancers(
    reactReduxFirebase(firebase, {
      userProfile: 'users',
      enableLogging: false
    })
  )(createStore);
  const store = createStoreWithMiddleware(rootReducer);

  // tslint:disable
  if ((module as any).hot) {
    // Enable Webpack hot module replacement for reducers
    (module as any).hot.accept('./reducers', () => {
      const nextRootReducer = require('./reducers');
      store.replaceReducer(nextRootReducer);
    });
  }
  // tslint:enable

  return store;
}
