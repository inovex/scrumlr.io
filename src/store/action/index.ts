import {BoardActionFactory, BoardAction, BoardReduxAction} from "./board";
import {NoteActionFactory, NoteAction, NoteReduxAction} from "./note";
import {VoteActionFactory, VoteAction, VoteReduxAction} from "./vote";
import {ParticipantAction, ParticipantActionFactory, ParticipantReduxAction} from "./participants";
import {ColumnActionFactory, ColumnAction, ColumnReduxAction} from "./column";
import {RequestActionFactory, RequestAction, RequestReduxAction} from "./request";
import {VotingActionFactory, VotingAction, VotingReduxAction} from "./votings";
import {AuthActionFactory, AuthAction, AuthReduxAction} from "./auth";
import {ViewAction, ViewActionFactory, ViewReduxAction} from "./view";
import {ReactionAction, ReactionActionFactory, ReactionReduxAction} from "./reaction";
import {BoardReactionAction, BoardReactionActionFactory, BoardReactionReduxAction} from "./boardReaction";
import {SkinToneAction, SkinToneActionFactory, SkinToneReduxAction} from "./skinTone";

/** This object lists all internal Redux Action types. */
export const Action = {
  ...BoardAction,
  ...ColumnAction,
  ...NoteAction,
  ...ReactionAction,
  ...AuthAction,
  ...ParticipantAction,
  ...RequestAction,
  ...VoteAction,
  ...VotingAction,
  ...ViewAction,
  ...BoardReactionAction,
  ...SkinToneAction,
};

/** Factory or creator class of internal Redux actions. */
export const Actions = {
  ...BoardActionFactory,
  ...ColumnActionFactory,
  ...NoteActionFactory,
  ...ReactionActionFactory,
  ...RequestActionFactory,
  ...AuthActionFactory,
  ...ParticipantActionFactory,
  ...RequestActionFactory,
  ...VoteActionFactory,
  ...VotingActionFactory,
  ...ViewActionFactory,
  ...BoardReactionActionFactory,
  ...SkinToneActionFactory,
};

/** The types of all application internal redux actions. */
export type ReduxAction = {context: {board?: string; user?: string; voting?: string; serverTimeOffset: number}} & (
  | BoardReduxAction
  | ColumnReduxAction
  | NoteReduxAction
  | ReactionReduxAction
  | AuthReduxAction
  | ParticipantReduxAction
  | RequestReduxAction
  | VoteReduxAction
  | VotingReduxAction
  | ViewReduxAction
  | BoardReactionReduxAction
  | SkinToneReduxAction
);
