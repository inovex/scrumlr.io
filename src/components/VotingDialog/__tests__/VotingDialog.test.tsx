import {fireEvent, render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import {Actions} from "store/action";
import store from "store";
import getTestStore from "utils/test/getTestStore";
import getTestApplicationState from "utils/test/getTestApplicationState";
import {ApplicationState} from "types";
import getTestParticipant from "utils/test/getTestParticipant";
import {I18nextProvider} from "react-i18next";
import {VotingDialog} from "..";
import i18nTest from "i18nTest";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

const storeDispatchSpy = jest.spyOn(store, "dispatch");

describe("VotingDialog", () => {
  const createVotingDialog = (overwrite?: Partial<ApplicationState>) => {
    return (
      <I18nextProvider i18n={i18nTest}>
        <Provider store={getTestStore(overwrite)}>
          <VotingDialog />
        </Provider>
      </I18nextProvider>
    );
  };

  beforeEach(() => {
    const portal = global.document.createElement("div");
    portal.setAttribute("id", "portal");
    global.document.querySelector("body")!.appendChild(portal);
  });

  it("should match the snapshot with no active voting", () => {
    const {container} = render(createVotingDialog({votings: {open: undefined, past: []}}), {container: global.document.querySelector("#portal")!});
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should match the snapshot with active voting", () => {
    const {container} = render(createVotingDialog(), {container: global.document.querySelector("#portal")!});
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should redirect if the current user isn't a moderator", () => {
    render(createVotingDialog({participants: {self: getTestParticipant({role: "PARTICIPANT"}), others: []}}), {container: global.document.querySelector("#portal")!});
    expect(mockedUsedNavigate).toHaveBeenCalledWith("..");
  });

  xit("should dispatch to store correctly on start voting button click", () => {
    render(createVotingDialog({votings: {open: undefined, past: []}}), {container: global.document.querySelector("#portal")!});
    fireEvent.click(screen.getByTestId("voting-dialog__start-button"));
    expect(storeDispatchSpy).toHaveBeenCalledWith(Actions.createVoting({voteLimit: 5, showVotesOfOthers: true, allowMultipleVotes: false}));
  });

  xit("should dispatch to store correctly on start voting button click with custom vote configuration", () => {
    render(createVotingDialog({votings: {open: undefined, past: []}}), {container: global.document.querySelector("#portal")!});
    fireEvent.click(screen.getByTestId("voting-dialog__cumulative-voting-button"));
    fireEvent.click(screen.getByTestId("voting-dialog__anonymous-voting-button"));
    fireEvent.click(screen.getByTestId("voting-dialog__plus-button"));
    fireEvent.click(screen.getByTestId("voting-dialog__start-button"));
    expect(storeDispatchSpy).toHaveBeenCalledWith(Actions.createVoting({voteLimit: 6, showVotesOfOthers: false, allowMultipleVotes: true}));
  });

  it("should dispatch to store correctly on cancel voting button click", () => {
    render(createVotingDialog(), {container: global.document.querySelector("#portal")!});
    fireEvent.click(screen.getByTestId("voting-dialog__cancel-button"));
    expect(storeDispatchSpy).toHaveBeenCalledWith(Actions.abortVoting(getTestApplicationState().votings.open!.id));
  });

  it("should dispatch to store correctly on stop voting button click", () => {
    render(createVotingDialog(), {container: global.document.querySelector("#portal")!});
    fireEvent.click(screen.getByTestId("voting-dialog__stop-button"));
    expect(storeDispatchSpy).toHaveBeenCalledWith(Actions.closeVoting(getTestApplicationState().votings.open!.id));
  });
});
