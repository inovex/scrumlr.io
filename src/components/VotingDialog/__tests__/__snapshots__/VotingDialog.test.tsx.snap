// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VotingDialog should match the snapshot with active voting 1`] = `
<div
  class="portal"
  role="dialog"
>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="portal__frame accent-color__backlog-blue"
    >
      <div
        class="portal__content"
        role="dialog"
      >
        <aside
          class="dialog voting-dialog accent-color__planning-pink"
        >
          <article
            class="dialog__content"
          >
            <h2
              class="dialog__header-text"
            >
              Voting
            </h2>
            <button
              class="voting-dialog__start-button"
              data-testid="voting-dialog__cancel-button"
            >
              <label>
                Cancel voting phase
              </label>
            </button>
            <button
              class="voting-dialog__start-button"
              data-testid="voting-dialog__stop-button"
            >
              <label>
                End voting phase
              </label>
            </button>
          </article>
          <button
            aria-label="Close"
            class="dialog__close-button"
          >
            <svg
              class="dialog__close-icon"
            >
              icon-close.svg
            </svg>
          </button>
        </aside>
      </div>
    </div>
  </div>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;

exports[`VotingDialog should match the snapshot with no active voting 1`] = `
<div
  class="portal"
  role="dialog"
>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="portal__frame accent-color__backlog-blue"
    >
      <div
        class="portal__content"
        role="dialog"
      >
        <aside
          class="dialog voting-dialog accent-color__planning-pink"
        >
          <article
            class="dialog__content"
          >
            <h2
              class="dialog__header-text"
            >
              Voting
            </h2>
            <button
              class="dialog__button"
              data-testid="voting-dialog__cumulative-voting-button"
            >
              <label>
                Allow cumulative voting
              </label>
              <div
                class="toggle toggle--active voting-dialog__toggle"
              />
            </button>
            <div
              class="dialog__button"
            >
              <label>
                Number of votes
              </label>
              <button
                aria-label="Decrease number of votes"
                class="voting-dialog__vote-button"
                data-testid="voting-dialog__minus-button"
              >
                <svg>
                  icon-minus.svg
                </svg>
              </button>
              <label
                class="voting-dialog__vote-label"
              >
                5
              </label>
              <button
                aria-label="Increase number of votes"
                class="voting-dialog__vote-button"
                data-testid="voting-dialog__plus-button"
              >
                <svg>
                  icon-plus.svg
                </svg>
              </button>
            </div>
            <button
              class="voting-dialog__start-button"
              data-testid="voting-dialog__start-button"
            >
              <label>
                Start voting phase
              </label>
            </button>
          </article>
          <button
            aria-label="Close"
            class="dialog__close-button"
          >
            <svg
              class="dialog__close-icon"
            >
              icon-close.svg
            </svg>
          </button>
        </aside>
      </div>
    </div>
  </div>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;
